transaction/builder/src/memo_builder/rth_memo_builder.rs

```diff
@@ -76,12 +75,44 @@ pub struct RTHMemoBuilder {
     fee: Amount,
 }
 
+#[derive(Clone, Debug)]
+pub enum CustomMemoType {
+    PaymentRequestId(u64),
+    PaymentIntentId(u64),
+    FlexibleMemos(FlexibleMemoPayloads),
+}
+
+/// This contains both the output memo payload and the change memo payload which
+/// will be used when generating custom output memos and change memos
+#[derive(Clone, Debug)]
+pub struct FlexibleMemoPayloads {
+    /// This is used when generating output memos.
+    /// It must contain a memo type with the first byte being 0x01
+    pub output_memo_payload: FlexibleMemoPayload,
+    /// This is used when generating change memos.
+    /// It must contain a memo type with the first byte being 0x02
+    pub change_memo_payload: FlexibleMemoPayload,
```

Comment:
1. This is too hard to use as an API. This was not even asked for by Moby team, so they probably don't even want this but now they have to construct it with the right type bytes. At the very least it should be an option that they can default to None.
2. There was not part of what was discussed in #3200 , I think it's just a scope creep and we shouldn't even do it.